pkg/manager/http.go

```diff
@@ -734,6 +739,48 @@ func (serv *HTTPServer) modulesInfo(w http.ResponseWriter, r *http.Request) {
 	serv.jsonPage(w, r, "modules", cover.Modules)
 }
 
+func (serv *HTTPServer) httpAddInput(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "multipart/form-data")
+	if r.Method != http.MethodPost {
+		http.Error(w, "only POST method supported", http.StatusMethodNotAllowed)
+		return
+	}
+	err := r.ParseMultipartForm(20 << 20)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to parse form: %v", err), http.StatusBadRequest)
+		return
+	}
+	file, _, err := r.FormFile("file")
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to retrieve file from form-data: %v", err), http.StatusBadRequest)
+		return
+	}
+	defer file.Close()
+	scanner := bufio.NewScanner(file)
+	var progData []byte
+	for scanner.Scan() {
+		data := scanner.Bytes()
+		progData = append(progData, data...)
+	}
+	target, err := prog.GetTarget(serv.Cfg.TargetOS, serv.Cfg.TargetArch)
```

Comment:
You can use `serv.Cfg.Target` which is already non-nil and does not need to be checked.